common/src/main/java/io/netty/util/concurrent/FastThreadLocal.java

```diff
@@ -190,7 +194,13 @@ private V initialize(InternalThreadLocalMap threadLocalMap) {
      */
     public final void set(V value) {
         if (value != InternalThreadLocalMap.UNSET) {
-            set(InternalThreadLocalMap.get(), value);
+            InternalThreadLocalMap threadLocalMap = InternalThreadLocalMap.get();
+            boolean alreadySet = threadLocalMap.isIndexedVariableSet(index);
```

Comment:
consider making an overload of `set` which returns `alreadySet` because it already knows this information. it could be `private static`.

Reply:
This is not alway true... let me just merge it as it is for now

Reply:
PR https://github.com/netty/netty/pull/7541 is what I had in mind